src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,86 @@
+package duke;
+
+import java.time.LocalDate;
+
+/**
+ * Takes in command from user and checks what to do with it
+ */
+public class Parser {
+    TaskList taskList = new TaskList();
+    Ui ui = new Ui();
+
+    Parser() {
+    }
+
+    /**
+     * Check if description is missing
+     *
+     * @param input
+     * @return validity
+     */
+    private boolean checkInvalidDescription(String input) {
+        if (input.trim().equals("todo") || input.trim().equals("deadline") || input.trim().equals("event")) {
+            return true;
+        }
+        return false;
+    }
+
+    public void load() {
+        taskList.load();
+    }
+
+    /**
+     * Take in input from user and checks what commands to run
+     *
+     * @param input from user
+     */
+    public String commands(String input) {
+        try {
+            if (input.equals("list")) { //checks for list
+                return taskList.printList();
```

Comment:
Perhaps can change the method name to something like readCommands or getCommands?